taxi/views.py

```diff
@@ -1,48 +1,54 @@
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
 from django.views import generic
 
 from .models import Driver, Car, Manufacturer
 
 
-
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
     num_drivers = Driver.objects.count()
     num_cars = Car.objects.count()
     num_manufacturers = Manufacturer.objects.count()
+    num_visit = request.session.get("num_visit", 1)
+    request.session["num_visit"] = num_visit + 1
+
 
     context = {
         "num_drivers": num_drivers,
         "num_cars": num_cars,
         "num_manufacturers": num_manufacturers,
+        "num_visit": num_visit,
```

Comment:
for now `num_visits` is displayed correctly, but when you fix a problem with `num_visits` in session you will have to fix `num_visits` in context to have the correct value

Reply:
set right